packages/kiota_abstractions/lib/src/enum_registry.dart

```diff
@@ -0,0 +1,49 @@
+part of '../kiota_abstractions.dart';
+
+/// This is a helper class to register [Enum] cases so they can be used in
```

Comment:
is this effectively so a "FooBar" symbol can be mapped with its "Foo BAR" string representation during serialization?

Reply:
Not quite. I'm currently writing some tests and found that this part is not yet working correctly (oops). The updated code will include a conversion method to convert enum cases to their string representation (because for some reason, Dart doesn't include the `name` getter in the `Enum` contract...)

Reply:
Found a way to achieve this. The `name` getter is just an extension method on the `Enum` type. So, casting to `Enum` first allows to get the `name` of the case.